instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -494,30 +561,51 @@ def __init__(
                 __name__,
                 __version__,
                 meter_provider,
-                schema_url="https://opentelemetry.io/schemas/1.11.0",
+                schema_url=_get_schema_url(sem_conv_opt_in_mode),
             )
             if meter is None
             else meter
         )
-        self.duration_histogram = self.meter.create_histogram(
-            name=MetricInstruments.HTTP_SERVER_DURATION,
-            unit="ms",
-            description="Duration of HTTP client requests.",
-        )
-        self.server_response_size_histogram = self.meter.create_histogram(
-            name=MetricInstruments.HTTP_SERVER_RESPONSE_SIZE,
-            unit="By",
-            description="measures the size of HTTP response messages (compressed).",
-        )
-        self.server_request_size_histogram = self.meter.create_histogram(
-            name=MetricInstruments.HTTP_SERVER_REQUEST_SIZE,
-            unit="By",
-            description="Measures the size of HTTP request messages (compressed).",
-        )
-        self.active_requests_counter = self.meter.create_up_down_counter(
-            name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,
-            unit="requests",
-            description="measures the number of concurrent HTTP requests that are currently in-flight",
+        self.duration_histogram_old = None
+        if _report_old(sem_conv_opt_in_mode):
+            self.duration_histogram_old = self.meter.create_histogram(
+                name=MetricInstruments.HTTP_SERVER_DURATION,
+                unit="ms",
+                description="Duration of HTTP server requests.",
+            )
+        self.duration_histogram_new = None
+        if _report_new(sem_conv_opt_in_mode):
+            self.duration_histogram_new = self.meter.create_histogram(
```

Comment:
can we use `semconv.create_http_server_request_duration` ?

Reply:
I am purposefully not using apis in semconv that are incubating. Could change in the future.

Reply:
Got it